ntpd/src/daemon/ntp_source.rs

```diff
@@ -137,55 +137,38 @@ where
                 },
             };
 
-            let actions = match selected {
-                SelectResult::Recv(result) => {
-                    tracing::debug!("accept packet");
-                    match accept_packet(result, &buf, &self.clock) {
-                        AcceptResult::Accept(packet, recv_timestamp) => {
-                            let send_timestamp = match self.last_send_timestamp {
-                                Some(ts) => ts,
-                                None => {
-                                    debug!(
-                                        "we received a message without having sent one; discarding"
```

Comment:
Needs fix: Again, this is likely (part of) a good refactor, but too hard to review well in this context, please split this off into a separate PR.